posts/2025-02-17_theres_a_pharmaverse_package_for_that/managing-clinical-trial-data.qmd

```diff
@@ -0,0 +1,323 @@
+  )
+```
+
+#### **Step 4: Assign Population Flags**
+
+For safety population (SAFFL), we check if the patient received a treatment dose:
+
+``` r
+adsl <- adsl %>%
+  derive_var_merged_exist_flag(
+    dataset_add = ex,
+    by_vars = exprs(STUDYID, USUBJID),
+    new_var = SAFFL,
+    condition = EXDOSE > 0 | str_detect(EXTRT, "PLACEBO")
+  )
+```
+
+#### **Step 5: Generate and Save Results**
+
+Finally, we print the results and save them to a CSV file:
+
+``` r
+tryCatch({
+  print("ADSL creation complete. Showing first few rows:")
+  print(head(adsl))
+  
+  print("\nSummary of ADSL variables:")
+  print(summary(adsl))
+  
+  # Save to a CSV file
+  write.csv(adsl, "adsl_output.csv", row.names = FALSE)
+  print("\nADSL has been saved to 'adsl_output.csv'")
+}, error = function(e) {
+  print(paste("An error occurred:", e$message))
+})
+```
+
+![](images/result1.png)
+
+![](images/result2.png)\
+
+#### **More Details on ADSL Creation**
+
+This is just a **high-level example**; the full process includes deriving death variables, grouping populations, and applying labels. For a deeper dive, check out the [ADSL Implementation Guide](https://cran.r-project.org/web/packages/admiral/vignettes/adsl.html).
+
+![](images/adsl.png)
+
+## **Who Are the Key Players in Pharmaverse, and Do You Need to Use All Packages?**
+
+[![Source: Getting Started with tidyverse in R](images/tidyverse-packages.png)](https://www.storybench.org/getting-started-with-tidyverse-in-r/)
+
+#### **Key Players in pharmaverse**
+
+-   **Pharma Industry Leaders** – Pharmaceutical companies, CROs, and regulatory bodies driving adoption.
+
+-   **Regulatory Agencies such as the FDA and EMA, and advisory agencies such as [CDISC](https://www.cdisc.org/), [PHUSE](https://phuse.global/))** – Setting standards for clinical data reporting.
+
+-   **Open-Source Community** – Developers and passionate individual contributors contributing to pharmaverse packages to improve functionality and compliance.
+
+-   **Organizations and Research Institutes** – Using pharmaverse for clinical trial data management.
+
+#### **Do You Need to Use All pharmaverse Packages?**
+
+-   No, organizations can select only the packages that fit their needs.
+
+-   Many packages are modular and independent, allowing selective integration.
+
+-   pharmaverse hosts multiple packages with similar aims, giving users the flexibility to choose what works best for them rather than prescribing a single approach.
+
+-   Some packages offer SAS-like functionality, easing transitions for SAS users.
+
+-   pharmaverse complements [Tidyverse](https://www.tidyverse.org/), allowing organizations to continue using familiar R workflows.
+
+## **How pharmaverse Differs from Tidyverse & How to Learn It Effectively**
+
+#### **Differences Between pharmaverse and Tidyverse**
+
+-   pharmaverse is designed specifically for clinical trial data management, while Tidyverse provides general-purpose data science tools.
```

Comment:
General comment: We usually use "clinical reporting" in pharmaverse site around our scope rather than "clinical trial data management" as it has slightly wider connotations